src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ClassWriterImpl.java

```diff
@@ -194,48 +188,48 @@ protected TypeElement getCurrentPageElement() {
     }
 
     @Override
-    public void addClassSignature(Content classInfoTree) {
-        classInfoTree.add(new HtmlTree(TagName.HR));
-        classInfoTree.add(new Signatures.TypeSignature(typeElement, this)
+    public void addClassSignature(Content classInfo) {
+        classInfo.add(new HtmlTree(TagName.HR));
+        classInfo.add(new Signatures.TypeSignature(typeElement, this)
                 .toContent());
     }
 
 
     @Override
-    public void addClassDescription(Content classInfoTree) {
-        addPreviewInfo(classInfoTree);
+    public void addClassDescription(Content classInfo) {
+        addPreviewInfo(classInfo);
         if (!options.noComment()) {
             // generate documentation for the class.
             if (!utils.getFullBody(typeElement).isEmpty()) {
-                addInlineComment(typeElement, classInfoTree);
+                addInlineComment(typeElement, classInfo);
             }
         }
     }
 
-    private void addPreviewInfo(Content classInfoTree) {
-        addPreviewInfo(typeElement, classInfoTree);
+    private void addPreviewInfo(Content content) {
+        addPreviewInfo(typeElement, content);
     }
 
     @Override
-    public void addClassTagInfo(Content classInfoTree) {
+    public void addClassTagInfo(Content classInfo) {
         if (!options.noComment()) {
             // Print Information about all the tags here
-            addTagsInfo(typeElement, classInfoTree);
+            addTagsInfo(typeElement, classInfo);
         }
     }
 
     /**
-     * Get the class hierarchy tree for the given class.
+     * Get the class inheritance tree for the given class.
```

Comment:
* Gets

"class inheritance" suggests superclass, which is not a tree. Depending on functional semantics, maybe just "class hierarchy", since superclasses do not form a tree and interfaces form an acyclic graph.

Reply:
I picked "inheritance" over "hierarchy" for consistency, because the relevant entities in HTML and CSS referred to "inheritance"; for example:

 * jdk/javadoc/internal/doclets/formats/html/resources/standard.properties:164
 * jdk/javadoc/internal/doclets/formats/html/markup/HtmlStyle.java:934

On the other hand, "hierarchy" was also a good choice because it could be used as a standalone noun, thus eliminating the need for the word "tree".

Thoughts?

Reply:
Yes, my criticism was about the use of the word "tree" in this case. While I agree that this is not the primary use you are trying to eliminate (i.e. HtmlTree-like trees) it  nevertheless is misleading to describe the superclasses of a class as forming a tree, since they form just a single branch of any tree.

Reply:
Is there a suitable term we could use to describe this _ancestry_ information? We are specifically interested in a superclass-subclass chain that connects a class in question with java.lang.Object.

Reply:
`class hierarchy` ?
`superclass chain` ?
`superclasses` ?
`inheritance chain` ?
`list of superclasses` ?
`ancestors` ?


'inheritance` isn't a bad word, it just doesn't work as a noun in this context. The problem is using it with `tree`, since the superclass-subclass chain is a degenerate tree that is a list.

Reply:
If we look at the terms available in JLS, what about "All Extended Classes" in jdk.javadoc codebase **and** the generated documentation? (Currently, "Inheritance Tree" is used as a tooltip title of the respective region.) "All Extended Classes" goes nicely with "All Implemented Interfaces". Alternatively we could use Superclasses and  Superinterfaces as per https://docs.oracle.com/javase/specs/jls/se18/html/jls-8.html#jls-8.1.4.

No matter which of those two we choose, it might feel awkward because of the inherent (no pun intended) asymmetry between classes and interfaces: there can be only one direct superclass but multiple direct superinterfaces.

Reply:
I think the distinction is between "down" from Object (subtypes form a tree) or "up" to Object (superclasses form a list, super interfaces form a graph)